utils/synapse_json_schema_bind.py

```diff
@@ -0,0 +1,194 @@
+        print(f"JSON schema {uri} was previously registered and will be bound to the entity.")
+        
+    return uri
+
+
+def bind_schema_to_entity(syn, service, schema_uri: str, entity_id: str, component_type: str):
+    """Associate a registered JSON schema with a Synapse entity.
+    For JSON schemas associated with DUO-based access restrictions, use the REST API and enable derived annotations,
+    For non-AR schemas, use the python client bind_json_schema function"""
+
+    if component_type != "duoCodeAR":
+        print(f"Binding non-AR schema {schema_uri}")
+        service.bind_json_schema(schema_uri, entity_id)
+
+    else:
+        print(f"Binding AR schema {schema_uri}")
+        request_body = {
+            "entityId": entity_id,
+            "schema$id": schema_uri,
+            "enableDerivedAnnotations": True
+            }
+        syn.restPUT(
+            f"/entity/{entity_id}/schema/binding", body=json.dumps(request_body)
+        )
+
+   
+def get_schema_from_url(component: str, version: str, grant: str) -> tuple[any, str]:
+    """Access a JSON schema stored in the MC2 Center data-models GitHub repo,
+    based on the version release tag, data type, and grant number (if of type duoCodeAR).
+    Return request JSON and adjusted component name.
+
+    Note that the filename must match expected conventions:
+    Non-AR schema example: mc2.DatasetView.schema.json
+    AR schema example: mc2.CA987654duoCodeAR.schema.json
+    """
+
+    #base_schema_url = "".join(["https://raw.githubusercontent.com/mc2-center/data-models/refs/tags/v", version, "/json_schemas/"])
+    base_schema_url = "".join(["https://raw.githubusercontent.com/mc2-center/data-models/refs/heads/136-173-dataset-schema/json_schemas/"])
+
+    if grant is not None and component == "duoCodeAR":
+        component = "".join([grant, component])
+    
+    component_json_name = ".".join(["mc2", component, "schema", "json"])
+    
+    schema_url = "".join([base_schema_url, component_json_name])
+
+    source_schema = requests.get(schema_url) 
+
+    schema_json = source_schema.json()
+
+    print(f"JSON schema {component} {version} successfully acquired from repository")
+
+    return schema_json, component
+
+
+def get_register_bind_schema(syn, component: str, grant: str, version: str, target: str, schema_org_name: str, org, service):
+    """Access JSON from URL, register the JSON schema, and bind the schema to the target entity."""
+    
+    schema_json, component_adjusted = get_schema_from_url(component, version, grant)
+    print(f"Registering JSON schema {component_adjusted} {version}")
+
+    uri = register_json_schema(org, component_adjusted, schema_json, version, schema_org_name)
+    
+    bound_schema = bind_schema_to_entity(syn, service, uri, target, component)
+    print(f"\nSchema {component_adjusted} {version} successfully bound to entity {target}")
+
+
+def main():
+
+    syn = (
+        synapseclient.Synapse()
+    )
+    syn.login()
```

Comment:
Similar to [`table_to_annotations.py`](https://github.com/mc2-center/mc2-center-dcc/blob/76a22c36dcf1599ae4199e7c57c33695a514dfa9/utils/table_to_annotations.py#L161), this could be simplified into a one-liner:

```python
syn = synapseclient.login()
```